### PR TITLE
choose different max_concurrent requests per drive based on HDD/NVMe

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
choose different max_concurrent requests per drive based on HDD/NVMe

## Motivation and Context
currently the default for all drives is 512, which is a lot 
for HDDs the recent testing has revealed moving this to 32 
for HDDs seems like a fair value.

## How to test this PR?
This should be automatically handled. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by poornas — APPROVED** [review]

File: cmd/xl-storage-disk-id-check.go [hunk 1]
```diff
@@ -777,7 +783,7 @@ type diskHealthTracker struct {
 }
 
 // newDiskHealthTracker creates a new disk health tracker.
-func newDiskHealthTracker() *diskHealthTracker {
+func newDiskHealthTracker(rotation bool, diskMaxConcurrent int) *diskHealthTracker {
```
Comment (vadmeste) [inline cmd/xl-storage-disk-id-check.go]: It doesn't seem rotation variable is used

File: cmd/xl-storage-disk-id-check.go [hunk 2]
```diff
@@ -81,11 +81,14 @@ type xlStorageDiskIDCheck struct {
...
-	diskID       string
-	storage      *xlStorage
-	health       *diskHealthTracker
+	apiCalls          [storageMetricLast]uint64
+	apiLatencies      [storageMetricLast]*lockedLastMinuteLatency
+	diskID            string
+	storage           *xlStorage
+	health            *diskHealthTracker
+	diskStartChecking int
+	diskMaxConcurrent int
```
Comment (vadmeste) [inline cmd/xl-storage-disk-id-check.go]: Let's add comments for those variables; I understand diskMaxConcurrent, but not diskStartChecking

**Review by vadmeste — COMMENTED** [review]
Few comments

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by klauspost** [comment]
Love it!

**Pull request merged**